src/java.base/share/classes/java/util/DualPivotQuicksort.java

```diff
@@ -269,10 +279,18 @@ static void sort(Sorter sorter, int[] a, int bits, int low, int high) {
             int upper = end; // The index of the first element of the right part
 
             /*
-             * Partitioning with 2 pivots in case of different elements.
+             * Partitioning with two pivots in case of different elements.
              */
             if (a[e1] < a[e2] && a[e2] < a[e3] && a[e3] < a[e4] && a[e4] < a[e5]) {
 
+                /*
+                 * Invoke radix sort on large array.
+                 */
```

Comment:
I prefer testing (sorter == null) first as it is always true for sequential sort and avoid testing bits > ... in this case

Reply:
It makes sense, I will update.